.gitlab-ci.yml

```diff
@@ -295,3 +324,25 @@ publish:aws-k8s-pipeline-toolbox:
     CONTAINER_TAG: "aws-k8s-v1"
   script:
     - *tag_n_push_to_gitlab_registry
+
+publish:mender-dist-packages-image:
+  extends: .template:publish
+  dependencies:
+    # NOTE: We should depend on each of the jobs individually with:
+    # - "build:mender-dist-packages-image: [${DISTRO}, ${RELEASE}, ${ARCH}]"
+    # However GitLab does not seem to expand these variables on a dependencies
+    # key. Use arbitrary one of them and assume that they all passed (using
+    # depends instead of needs ensures that the whole previous build stage
+    # needs to pass, so it should be safe).
+    - "build:mender-dist-packages-image: [debian, bullseye, amd64]"
```

Comment:
on my short look into the parallel feature I haven't seen if just the expansion doesn't work or the entire "depending on a resulting job name" doesn't work, but I would expect this to work and the comment is very valuable 🙏!

Reply:
I made some experiments. I think it is the former (the expansion is the problem). [This code](https://github.com/mendersoftware/mender-test-containers/compare/b075b6e78d63c1aebb4b3912c32167a02cdf9897..167b2dfc084237457d1c5c3cc9660d831c4fbc29) worked, but [this one](https://github.com/mendersoftware/mender-test-containers/compare/167b2dfc084237457d1c5c3cc9660d831c4fbc29..f60055998bd07d24494e2cd33984867ea8ae8b9a) didn't.